src/sentry/scripts/similarity/index.lua

```diff
@@ -0,0 +1,618 @@
+                                        get_bucket_frequency_key(
+                                            configuration.scope,
+                                            index,
+                                            time,
+                                            band,
+                                            key
+                                        )
+                                    )
+                                end
+                            ),
+                            function (result, response)
+                                for i = 1, #response, 2 do
+                                    local bucket, count = unpack(table.slice(response, i, i + 1))
+                                    result[bucket] = (result[bucket] or 0) + count
+                                end
+                                return result
+                            end,
+                            {}
+                        )
+                    end
+                )
+            end
+
+            local fetch_candidates = function (index, frequencies)
+                local candidates = {}  -- acts as a set
+                for band, buckets in ipairs(frequencies) do
+                    for bucket, count in pairs(buckets) do
+                        for _, time in ipairs(time_series) do
+                            -- Fetch all other items that have been
+                            -- added to the same bucket in this bad
+                            -- during this time period.
+                            local members = redis.call(
+                                'SMEMBERS',
+                                get_bucket_membership_key(
+                                    configuration.scope,
+                                    index,
+                                    time,
+                                    band,
+                                    bucket
+                                )
+                            )
+                            for _, member in ipairs(members) do
+                                candidates[member] = true
+                            end
+                        end
+                    end
+                end
+                return candidates
+            end
+
+            return table.imap(
+                indices,
+                function (index)
+                    local item_frequencies = fetch_bucket_frequencies(index, item_key)
+
+                    local candidates = fetch_candidates(index, item_frequencies)
+                    local candidate_frequencies = {}
+                    for candidate_key, _ in pairs(candidates) do
+                        candidate_frequencies[candidate_key] = fetch_bucket_frequencies(
+                            index,
+                            candidate_key
+                        )
+                    end
+
+                    local results = {}
+                    for key, value in pairs(candidate_frequencies) do
```

Comment:
maybe throw a couple comments on what/how we're computing in this loop - reads fine now since i'm fresh on it but worried about someone else or 6 months ago